### PR TITLE
[msal-browser] Adding exports and other changes for extensibility

### PR DESCRIPTION
**Pull request by pkanher617**
This PR adds additional exported files and other changes to further support extensibility in other libraries.

### CODE REVIEW

File: lib/msal-browser/src/interaction_client/PopupClient.ts [hunk 1]
```diff
@@ -189,7 +190,7 @@ export class PopupClient extends StandardInteractionClient {
                 const absoluteUrl = UrlString.getAbsoluteUrl(mainWindowRedirectUri, BrowserUtils.getCurrentUri());
 
                 this.logger.verbose("Redirecting main window to url specified in the request");
-                this.logger.verbosePii(`Redirecting main window to: ${absoluteUrl}`);
+                this.logger.verbosePii(`Redirecing main window to: ${absoluteUrl}`);
```
Comment (tnorling) [inline lib/msal-browser/src/interaction_client/PopupClient.ts]: Regression?

File: lib/msal-browser/src/interaction_client/RedirectClient.ts [hunk 2]
```diff
@@ -83,6 +84,7 @@ export class RedirectClient extends StandardInteractionClient {
             let state: string;
             try {
                 state = this.validateAndExtractStateFromHash(responseHash, InteractionType.Redirect);
+                BrowserUtils.clearHash(window);
```
Comment (tnorling) [inline lib/msal-browser/src/interaction_client/RedirectClient.ts]: Also regression?

Comment (peterzenz) [reply lib/msal-browser/src/interaction_client/RedirectClient.ts]: For this and line 25 above, neither of these fit with the visibility changes (private/protected) that you have in the rest of the PR.  Is this incorporating a change from a different commit/PR that you've brought in?

File: lib/msal-browser/src/interaction_client/StandardInteractionClient.ts [hunk 3]
```diff
@@ -250,8 +249,6 @@ export abstract class StandardInteractionClient extends BaseInteractionClient {
             }
         }
 
-        this.browserStorage.updateCacheEntries(validatedRequest.state, validatedRequest.nonce, validatedRequest.authority, validatedRequest.loginHint || "", validatedRequest.account || null);
```
Comment (peterzenz) [inline lib/msal-browser/src/interaction_client/StandardInteractionClient.ts:273]: Changes in this file, like the RedirectClient.ts look unrelated to the intent of the PR, maybe you need to re-sync with other updates.

File: lib/msal-browser/src/utils/BrowserConstants.ts [hunk 4]
```diff
@@ -106,7 +106,8 @@ export enum ApiId {
 export enum InteractionType {
     Redirect = "redirect",
     Popup = "popup",
-    Silent = "silent"
+    Silent = "silent",
+    None = "none"
```
Comment (peterzenz) [inline lib/msal-browser/src/utils/BrowserConstants.ts:110]: This change also looks unrelated to the intent of your PR.

**Review by peterzenz — APPROVED** [review]
Please confirm the changes that don't match the extensibility changes are not regressions before committing.

File: lib/msal-browser/src/interaction_client/RedirectClient.ts (hunk 2 above)
Comment (pkanher617) [reply lib/msal-browser/src/interaction_client/RedirectClient.ts]: These are merge conflicts from PRs that were merged after this branch was created, resolving them now.

Comment (pkanher617) [reply lib/msal-browser/src/interaction_client/RedirectClient.ts]: Also, the cache changes are meant to be done, as they are part of the "extensibility" changes

File: lib/msal-browser/src/interaction_client/StandardInteractionClient.ts (hunk 3 above)
Comment (pkanher617) [reply lib/msal-browser/src/interaction_client/StandardInteractionClient.ts:273]: updating cache entries is part of the "extensiblity" changes, those are now done in their respective interaction handlers

File: lib/msal-browser/src/utils/BrowserConstants.ts (hunk 4 above)
Comment (pkanher617) [reply lib/msal-browser/src/utils/BrowserConstants.ts:110]: also related to enums available for downstream packages

Comment (tnorling) [inline lib/msal-browser/src/utils/BrowserConstants.ts:110]: Did you get a chance to see if this could be added in the 1p lib instead of here?

File: lib/msal-browser/src/interaction_client/SilentIframeClient.ts [hunk 5]
```diff
@@ -44,6 +44,7 @@ export class SilentIframeClient extends StandardInteractionClient {
             ...request,
             prompt: PromptValue.NONE
         }, InteractionType.Silent);
+        this.browserStorage.updateCacheEntries(silentRequest.state, silentRequest.nonce, silentRequest.authority, silentRequest.loginHint || "", silentRequest.account || null);
```
Comment (tnorling) [inline lib/msal-browser/src/interaction_client/SilentIframeClient.ts:47]: Can we put this in `initializeAuthorizationRequest`?

Can you also explain what this is for?

Comment (pkanher617) [reply lib/msal-browser/src/interaction_client/SilentIframeClient.ts:47]: This is to update temporary cache values. Using InitializeAuthorizationRequest makes it harder to control when the cache values are updated. For some broker scenarios, we don't want to update cache entries when we initialize the request, but after we send the request up to the broker.

File: lib/msal-browser/src/utils/BrowserConstants.ts (hunk 4 above)
Comment (pkanher617) [reply lib/msal-browser/src/utils/BrowserConstants.ts:110]: Yes, you can't extend enums and it wouldn't make sense to add as a separate enum.

File: lib/msal-browser/src/interaction_client/SilentIframeClient.ts (hunk 5 above)
Comment (tnorling) [reply lib/msal-browser/src/interaction_client/SilentIframeClient.ts:47]: I see now that's where it used to be... Can we add a comment explaining this? This feels like something that may get moved back at some point without realizing what it affects. Not sure I like taking a dependency on implementation details like this

**Review by tnorling — APPROVED** [review]

File: lib/msal-browser/src/interaction_client/SilentIframeClient.ts (hunk 5 above)
Comment (sameerag) [reply lib/msal-browser/src/interaction_client/SilentIframeClient.ts:47]: I have the same inclination that this may get "optimized" at some point. 

Least case add a comment, Best case make a commented helper API with instructions regarding the sequence , 
`processReqBeforeDispatch(browserStorage, request: BaseReqType)`.

May prevent someone moving this later that way and any other functionality we may add before request dispatch for broker. Could be an overkill, will let you do the trade off.

Comment (sameerag) [inline lib/msal-browser/src/interaction_client/SilentIframeClient.ts:47]: nit: Should we replace "" with `Constants.EMPTY_STRING`?

**Review by sameerag — APPROVED** [review]
Approved with comments
